core/src/smartcontracts/isi/triggers/mod.rs

```diff
@@ -91,12 +91,15 @@ pub mod isi {
                             .map_err(|e: &str| Error::Conversion(e.to_owned()))?,
                     )
                 }
-                TriggeringEventFilterBox::ExecuteTrigger(_) => triggers.add_by_call_trigger(
+                EventFilterBox::ExecuteTrigger(_) => triggers.add_by_call_trigger(
                     &engine,
                     new_trigger
                         .try_into()
                         .map_err(|e: &str| Error::Conversion(e.to_owned()))?,
                 ),
+                EventFilterBox::TriggerCompleted(_) => {
+                    unreachable!("Disallowed during deserialization");
```

Comment:
Maybe return error explicitly? Because if we would do some kind of unit test and construct `EventFilterBox` by hand it won't be rejected during deserialization.

Reply:
test case will then fail, won't it?

Reply:
hmm, indeed
my thought was that error is easier to check